### PR TITLE
[GUI] One-click export to file

### LINKED ISSUE #1417: [Feature request] Removal of unncessary extra mouse-click while creating wallet backup
Currently, user is asked to select folder where he/she wants to create PIVX backup wallet file. User has to type in the file name and by clicking "Save", it has to click again "Backup" button in wallet. 
Is it possible to create backup file automatically after clicking "Save" button (or however else it is called in Linux/MacOS), and removing that unnecessary extra step of clicking "Backup" button?

instantBackup

### PR DESCRIPTION
**Pull request by random-zebra**
Export to file directly, after selecting the destination and clicking "Save", without requiring additional button presses.
Interested flows:
- backup wallet
- export txes to csv
- export addresses to csv

Closes #1417

### CODE REVIEW

File: src/qt/pivx/settings/settingsexportcsv.cpp [hunk 1]
```diff
@@ -55,32 +51,38 @@ SettingsExportCSV::SettingsExportCSV(PIVXGUI* _window, QWidget *parent) :
     setFilterAddressBook(ui->comboBoxSortAddressType, lineEditAddressBook);
     ui->comboBoxSortAddressType->setCurrentIndex(0);
 
-    connect(ui->pushButtonSave, SIGNAL(clicked()), this, SLOT(exportClicked()));
     connect(ui->pushButtonDocuments, &QPushButton::clicked, [this](){selectFileOutput(true);});
+    connect(this, &SettingsExportCSV::fileSelectedTxes, [this](QString filename){exportTxes(filename);});
```
Comment (Fuzzbawls) [inline src/qt/pivx/settings/settingsexportcsv.cpp]: No need for a lambda here, can change to
```cpp
connect(this, &SettingsExportCSV::fileSelectedTxes, this, &SettingsExportCSV::exportTxes);
```

File: src/qt/pivx/settings/settingsexportcsv.cpp [hunk 2]
```diff
@@ -55,32 +51,38 @@ SettingsExportCSV::SettingsExportCSV(PIVXGUI* _window, QWidget *parent) :
...
 
-    connect(ui->pushButtonExportAddress, SIGNAL(clicked()), this, SLOT(onExportAddressesClicked()));
     connect(ui->pushButtonAddressDocuments, &QPushButton::clicked, [this](){selectFileOutput(false);});
+    connect(this, &SettingsExportCSV::fileSelectedAddrs, [this](QString filename){exportAddrs(filename);});
```
Comment (Fuzzbawls) [inline src/qt/pivx/settings/settingsexportcsv.cpp]: No need for a lambda here, can change to
```cpp
connect(this, &SettingsExportCSV::fileSelectedAddrs, this, &SettingsExportCSV::exportAddrs);
```

File: src/qt/pivx/settings/settingsexportcsv.cpp (hunk 1 above)
Comment (random-zebra) [reply src/qt/pivx/settings/settingsexportcsv.cpp]: yes, will update this once #1049 is merged.
This will need a rebase anyway

**Comment by random-zebra** [comment]
Rebased.

File: src/qt/pivx/settings/settingsexportcsv.cpp [hunk 3]
```diff
@@ -55,32 +51,38 @@ SettingsExportCSV::SettingsExportCSV(PIVXGUI* _window, QWidget *parent) :
...
                                         isTxExport ? tr("PIVX_tx_csv_export(*.csv)") : tr("PIVX_addresses_csv_export(*.csv)"),
                                         nullptr);
 
-    if (!filenameRet.isEmpty()) {
-        if (isTxExport) {
-            filename = filenameRet;
+    if (isTxExport) {
+        if (!filename.isEmpty()) {
             ui->pushButtonDocuments->setText(filename);
+            Q_EMIT fileSelectedTxes(filename);
```
Comment (furszy) [inline src/qt/pivx/settings/settingsexportcsv.cpp]: Remember that you are in the same object running in the same thread, no need to create a signal/slot connection. The method can be called directly.

Comment (random-zebra) [reply src/qt/pivx/settings/settingsexportcsv.cpp]: good point

**Comment by random-zebra** [comment]
Removed un-needed signal/slot connection in `SettingsExportCSV`.

**Pull request merged**